.github/workflows/promote.yaml

```diff
@@ -0,0 +1,43 @@
+name: Promote Charm
+
+on:
```

Comment:
I like this workflow file. I've actually never seen it before. 

Question: is anyone authorised to run this workflow? Might be good to set this to management / senior eng

Reply:
Everyone in the team yes as the whole team is admin on this github project, but this is configurable through github permission.